client/packages/system/src/Encounter/DetailView/useSaveWithStatusChange.tsx

```diff
@@ -56,7 +60,11 @@ export function useSaveWithStatusChange(
       onSave();
       return;
     }
-    updateEncounter({ status });
+    if (status === EncounterNodeStatus.Visited) {
+      updateEncounter({ status, endDatetime });
+    } else {
+      updateEncounter({ endDatetime });
```

Comment:
```suggestion
      updateEncounter({ status });
```

Sorry, this was my bad!